rmw_connext_shared_cpp/src/node_info_and_types.cpp

```diff
@@ -0,0 +1,289 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string.h>
+
+#include <functional>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+#include "rcutils/allocator.h"
+#include "rcutils/error_handling.h"
+#include "rcutils/logging_macros.h"
+#include "rcutils/strdup.h"
+#include "rcutils/types.h"
+
+#include "rmw/allocators.h"
+#include "rmw/convert_rcutils_ret_to_rmw_ret.h"
+#include "rmw/error_handling.h"
+#include "rmw/get_topic_names_and_types.h"
+#include "rmw/impl/cpp/macros.hpp"
+#include "rmw/impl/cpp/key_value.hpp"
+#include "rmw/names_and_types.h"
+#include "rmw/rmw.h"
+
+#include "rmw_connext_shared_cpp/types.hpp"
```

Comment:
For reasons I do not yet understand, macOS throws tons and tons of warnings on this include: https://ci.ros2.org/job/ci_osx/4759/consoleFull, search for "storage class identifier is deprecated".

Reply:
These warnings are in Connext itself, so we suppress them in other files which include these headers, e.g.:

https://github.com/ros2/rmw_connext/blob/e8e50e12ea4a8f48201fe6947ed1cb7df8acce22/rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/ndds_include.hpp#L18-L30

Reply:
Ah great, thanks.  I've fixed it in 7f7ae2d.  I still need a fix for cxx11 problem in rmw_fastrtps before I can run CI, but I'll follow up over there.